CorsixTH/CorsixTH
# [RFC] Add GoogleTest framework and some unit tests

This PR is large, however the majority is unit tests as I didn't want to add a testing framework then leave it unused.

Building on the feedback in #1318 this PR:

- Adds Google Testing Framework (we clone and build rather than link, explanation in #1318 )
- Bumps us to using C++-14 standard.
- Makes the unit test target optional and opt-in
- Enables CPP Tests for Travis builds
- Adds unit tests for `THChunkRenderer` 

---

To test:
- Check that by default unit tests are enabled
- Try enabling unit tests (the var is called `ENABLE_UNIT_TESTS` ) and check it correctly runs CMake
- Check the building `CorsixTH` does not build unit tests
- Check doing a bare build does
- Move into $BuildDir/CorsixTH
- Run `ctest -VV` check all tests pass correctly
- Code review.

As always feel free to suggest any changes / improvements ideas.
